### PR TITLE
Updates for regenesis 10/10/22

### PR DESCRIPTION
**Pull request by uF4No**
Updates the dependencies for 10th Oct regenesis.

New Docker image v1.2.0
New NPM dependencies
Fixes for errors identified in contracts (reviewed with @StanislavBreadless)

### CODE REVIEW

File: contracts/TwoUserMultisig.sol [hunk 1]
```diff
@@ -1,74 +1,132 @@
...
+    function executeTransaction(
+        bytes32,
+        bytes32,
+        Transaction calldata _transaction
+    ) external payable override onlyBootloader {
+        _executeTransaction(_transaction);
+    }
 
-	function _executeTransaction(Transaction calldata _transaction) internal {
+    function _executeTransaction(Transaction calldata _transaction) internal {
```
Comment (StanislavBreadless) [inline contracts/TwoUserMultisig.sol]: In the original tutorial the following code is mentioned:

```ts
function _execute(Transaction calldata _transaction) internal {
    address to = address(uint160(_transaction.to));
    uint256 value = _transaction.reserved[1];
    bytes memory data = _transaction.data;

    if(to == address(DEPLOYER_SYSTEM_CONTRACT)) {
        // We allow calling ContractDeployer with any calldata
        SystemContractsCaller.systemCall(
            uint32(gasleft()),
            to,
            uint128(_transaction.reserved[1]), // By convention, reserved[1] is `value`
            _transaction.data
        );
    } else {
        bool success;
        assembly {
            success := call(gas(), to, value, add(data, 0x20), mload(data), 0, 0)
        }
        require(success);
    }
}
```

Could you please try using this one?

File: deploy/deploy-multisig.ts [hunk 2]
```diff
@@ -1,47 +1,57 @@
...
 // Put the address of your AA factory
-const AA_FACTORY_ADDRESS = "0x4B5DF730c2e6b28E17013A1485E5d9BC41Efe021";
+const AA_FACTORY_ADDRESS = '<YOUR_FACTORY_ADDRESS>'
 
 export default async function (hre: HardhatRuntimeEnvironment) {
-  const provider = new Provider(hre.config.zkSyncDeploy.zkSyncNetwork);
-  const wallet = (new Wallet("<PRIVATE-KEY>")).connect(provider);
-  const factoryArtifact = await hre.artifacts.readArtifact("AAFactory");
+  const provider = new Provider(hre.config.zkSyncDeploy.zkSyncNetwork)
+  const wallet = new Wallet('<YOUR_PRIVATE_KEY>').connect(provider)
```
Comment (StanislavBreadless) [inline deploy/deploy-multisig.ts]: Let's leave correcting it for some other time, but we usually do use ";" at the end of the line and it is generally considered a good practice

Comment (uF4No) [reply deploy/deploy-multisig.ts]: yeah, we gotta have a consistent codebase so we should consider sharing VSCode/Prettier/ESLint settings (I'm a single quote guy 😅). Let's merge this now and solve those styling things in the next PRs

File: contracts/TwoUserMultisig.sol (hunk 1 above)
Comment (uF4No) [reply contracts/TwoUserMultisig.sol]: Fixed in 55fbeee3b909bb5e0a907a9efc0deea0f67ec8f6

**Review by StanislavBreadless — APPROVED** [review]

**Pull request merged**